Automattic/wp-calypso
# Make bundled theme generic (pre checkout)

Related to https://github.com/Automattic/themes/issues/7498

## Proposed Changes

* It's a refactor to make the bundle theme more generic, so we can easily add other themes on the future. Basically, all the settings are part of the new `client/my-sites/theme/bundle-settings.tsx`.
  * This doesn't include the installation flow yet. I decided to explore it as a separate thing and raise another PR to avoid more complexity in this one.
* ~~I also added the "[Status] String Freeze" label to this PR, so we make sure we don't break the translations of strings that are already in the site but had their structure changed.~~ https://github.com/Automattic/wp-calypso/pull/84483#issuecomment-1828504748

## Testing Instructions



Themes

* Navigate to http://calypso.localhost:3000/themes
* Select a site.
* Search for "Tsubaki".
* Make sure it continues working as properly, as previously.

Design Picker and checkout

* Navigate to the http://calypso.localhost:3000/start
* Navigate through the flow until get to the Design Picker.
* Filter by store.
* Check the Tsubaki theme and make sure it continues working properly, as previously.
* Choose the theme, click on "Upgrade Plan", and make sure it goes to the checkout to buy the business plan.
* Complete the checkout, and check that it returns to the theme and that you can continue.

## Screenshots

**Themes**

<img width="709" alt="Screenshot 2023-11-27 at 16 06 43" src="https://github.com/Automattic/wp-calypso/assets/876340/dfee8608-4da1-4acb-afec-a485378753f9">
<img width="803" alt="Screenshot 2023-11-27 at 16 06 59" src="https://github.com/Automattic/wp-calypso/assets/876340/4b2a7cf6-bf7a-4042-a764-c28b0c135fe6">

**Design Picker**

<img width="1025" alt="Screenshot 2023-11-27 at 16 05 56" src="https://github.com/Automattic/wp-calypso/assets/876340/1863223b-2371-4e75-8fa4-9975f98542d5">
<img width="863" alt="Screenshot 2023-11-27 at 16 06 12" src="https://github.com/Automattic/wp-calypso/assets/876340/56864fe5-ffdf-4286-a4c9-edfcb923ea02">

## Pre-merge Checklist




About the first item, I didn't test exhaustively the browser support because it shouldn't change anything for the browser.
